buildtrigger/customhandler.py

```diff
@@ -84,7 +84,7 @@ class CustomBuildTrigger(BuildTriggerHandler):
             "ref": {
                 "type": "string",
                 "description": "git reference for a git commit",
-                "pattern": "^refs\/(heads|tags|remotes)\/(.+)$",
+                "pattern": "^refs/(heads|tags|remotes)/(.+)$",
```

Comment:
note that python interprets `"foo\/bar"` as `"foo\\/bar"`, so maybe it needs to be `r"^refs\/(heads|tags|remotes)\/(.+)$"`

Reply:
I've tested both `r"^refs\/(heads|tags|remotes)\/(.+)$"` and `r"^refs/(heads|tags|remotes)/(.+)$"` and they seem to work the same way.

```python
import jsonschema
from pprint import pprint

schema = {
    "type": "object",
    "properties": {
        "ref": {
            "type": "string",
            "description": "git reference for a git commit",
            "pattern": r"^refs\/(heads|tags|remotes)\/(.+)$",
        },
    },
}

def check(obj, validator):
    pprint(obj)
    result = "VALID" if validator.is_valid(obj) else "INVALID"
    print(f"=> {result}")

validator = jsonschema.Draft4Validator(schema)
check({"ref": "refs/heads/master"}, validator)
# valid
check({"ref": "refs/tags"}, validator)
# invalid

# now without the escape
schema = {
    "type": "object",
    "properties": {
        "ref": {
            "type": "string",
            "description": "git reference for a git commit",
            "pattern": r"^refs/(heads|tags|remotes)/(.+)$",
        },
    },
}

validator = jsonschema.Draft4Validator(schema)
check({"ref": "refs/heads/master"}, validator)
# valid
check({"ref": "refs/tags"}, validator)
# invalid
```

Reply:
nice!